back/app/controllers/web_api/v1/project_reviews_controller.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+class WebApi::V1::ProjectReviewsController < ApplicationController
+  def show
+    if review
+      authorize(review)
+      render json: serialize(review)
+    else
+      authorize(project, :update?)
+      head(:no_content)
+    end
+  end
+
+  def create
+    review = ProjectReview.new(
+      project_id: params[:project_id],
+      requester: current_user,
+      reviewer_id: params[:project_review][:reviewer_id]
+    )
+
+    authorize(review)
+
+    side_fx.before_create(review, current_user)
+    if review.save
+      side_fx.after_create(review, current_user)
+      render json: serialize(review), status: :created
+    else
+      render json: { errors: review.errors }, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    if review
+      if params[:project_review][:state] == 'approved' && !review.approved?
+        review.approve(current_user)
+        authorize(review)
+        unless review.valid?
+          render json: { errors: review.errors }, status: :unprocessable_entity
+          return
+        end
+
+        side_fx.before_update(review, current_user)
+        review.save!
+        side_fx.after_update(review, current_user)
+      end
+
+      render json: serialize(review)
+    else
+      authorize(project, :update?)
+      head(:not_found)
+    end
+  end
+
+  def destroy
```

Comment:
Is the idea here to be able to reject a review? I think it's easy to do but we don't have a design for it. Should we sync with Rob?